universal-login-sdk/lib/core/services/TransferService.ts

```diff
@@ -16,13 +21,34 @@ export class TransferService {
   }
 
   validateInputs(transferDetails: TransferDetails, balance: Nullable<string>) {
+    this.errors = {amount: [], to: []};
     const {gasPrice, gasToken} = transferDetails.gasParameters;
     const gasCostInWei = utils.bigNumberify(DEFAULT_GAS_LIMIT.toString()).mul(gasPrice);
     ensureNotNull(balance, Error, 'Balance is null');
     const isAmountValid = isValidAmount(transferDetails.transferToken, balance, gasToken, gasCostInWei, transferDetails.amount);
+    if (!isAmountValid) {
+      this.errors.amount.push(`Amount ${transferDetails.amount} is not valid`);
+    }
     const isRecipientValid = isValidRecipient(transferDetails.to);
-    ensure(isAmountValid || isRecipientValid, InvalidAmountAndRecipient, transferDetails.amount, transferDetails.to);
-    ensure(isAmountValid, InvalidAmount, transferDetails.amount);
-    ensure(isRecipientValid, InvalidAddressOrEnsName, transferDetails.to);
+    if (!isRecipientValid) {
+      this.errors.to.push(`Recipient ${transferDetails.to} is not valid`);
+    }
+    return this.errors;
+  }
+
+  areInputsValid() {
+    return this.errors.amount.length === 0 && this.errors.to.length === 0;
+  }
+
+  getMaxAmount(gasParameters: GasParameters, balance: Nullable<string>) {
+    ensureNotNull(balance, Error, 'Balance is null');
+    const {gasPrice, gasToken} = gasParameters;
+    if (gasToken !== ETHER_NATIVE_TOKEN.address) {
+      return balance;
+    }
+    const gasCostInWei = utils.bigNumberify(DEFAULT_GAS_LIMIT.toString()).mul(gasPrice);
+    const maxAmountAsBigNumber = utils.parseEther(balance).sub(gasCostInWei);
+    const maxAmountValidated = isBiggerThan(maxAmountAsBigNumber, utils.parseEther('0'));
```

Comment:
`utils.parseEther('0')` is still 0.

Reply:
But 0 is not BigNumber which is required by isBiggerThan (getBigger).